src/main/java/org/jenkinsci/plugins/workflow/multibranch/DurabilityHintBranchProperty.java

```diff
@@ -39,44 +45,63 @@ public DurabilityHintBranchProperty(@Nonnull FlowDurabilityHint hint) {
         this.hint = hint;
     }
 
+    /** No-op impl because we only care about the actual BranchProperty attached. */
     @Override
     public final <P extends Job<P, B>, B extends Run<P, B>> JobDecorator<P, B> jobDecorator(Class<P> clazz) {
-        if (!clazz.isAssignableFrom(WorkflowJob.class)) {
-            return null;
-        }
-        return new JobDecorator<P, B>() {
-            @NonNull
-            @Override
-            public List<JobProperty<? super P>> jobProperties(
-                    @NonNull List<JobProperty<? super P>> jobProperties) {
-                List<JobProperty<? super P>> result = asArrayList(jobProperties);
-                for (Iterator<JobProperty<? super P>> iterator = result.iterator(); iterator.hasNext(); ) {
-                    JobProperty<? super P> p = iterator.next();
-                    if (p instanceof DurabilityHintJobProperty) {
-                        iterator.remove();
-                    }
-                }
-                if (getHint() != null) {
-                    result.add((JobProperty)(new DurabilityHintJobProperty(getHint())));
-                }
-                return result;
-            }
-        };
+        return null;
     }
 
     @Extension
-    public static class DescriptorImpl extends BranchPropertyDescriptor {
+    public static class DescriptorImpl extends BranchPropertyDescriptor implements DurabilityHintProvider {
         @Override
         public String getDisplayName() {
-            return "Pipeline speed/durability override";
+            return "Pipeline branch speed/durability override";
         }
 
         public FlowDurabilityHint[] getDurabilityHintValues() {
             return FlowDurabilityHint.values();
         }
 
-        public static FlowDurabilityHint getDefaultHint() {
+        public static FlowDurabilityHint getDefaultDurabilityHint() {
             return GlobalDefaultFlowDurabilityLevel.getDefaultDurabilityHint();
         }
+
+        @Override
+        /** Lower ordinal than {@link org.jenkinsci.plugins.workflow.job.properties.DurabilityHintJobProperty} so those can override. */
+        public int ordinal() {
+            return 200;
+        }
+
+        @CheckForNull
+        @Override
+        /**
+         * Dynamically fetch the property with each build, because the {@link BranchPropertyStrategy} does not re-evaluate,
```

Comment:
I suppose the issue here is that `BranchProjectFactory.decorate` is called only when the branch project is created, not after the `MultiBranchProject` is reconfigured? If so, smells like a bug in `branch-api` to me. As mentioned in https://github.com/jenkinsci/workflow-job-plugin/pull/75, branch properties have been nearly unused so far.

Anyway fetching on the fly is fine (certainly prefer to modifying the job `config.xml` when that is avoidable).